src/main/java/seedu/address/ui/CommandHistory.java

```diff
@@ -0,0 +1,103 @@
+package seedu.address.ui;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.logging.Logger;
+
+import seedu.address.commons.core.LogsCenter;
+
+/**
+ * Keeps track of which commands have been previously entered, so that past commands
+ * can easily be re-entered.
+ *
+ * Follows a singleton design pattern, so that the command history can only be instantiated once.
+ */
+public class CommandHistory {
+
+    private static CommandHistory commandHistory = null;
+
+    private final Logger logger = LogsCenter.getLogger(CommandHistory.class);
+    private final List<String> history;
```

Comment:
Collection renaming

Reply:
Fixed in https://github.com/AY2021S1-CS2103T-F11-1/tp/pull/193/commits/7b51604368f27be8ef423c281ab1f0faea76cf08